File: cp/graphs/bfs.cpp
#include <bits/stdc++.h>

using namespace std;

vector< list<int> > graph(5);

void bfs(){
    
}

int main()
{
    set<int> visited;
    set<int>::iterator it;
    graph[0] = {1,2,3};
    graph[1] = {0,3,4};
    graph[2] = {0,3};
    graph[3] = {0,1,2};
    graph[4] = {1,2};

    return 0;
}

File: cp/graphs/bfs.py
graph = {
    'A':['B', 'C','D'],
    'B':['A','D','E'],
    'C':['A','D'],
    'D':['A','B','C'],
    'E':['B','D']
}

queue = []
visited = []
def bfs(visited, graph, node):
  visited.append(node)
  queue.append(node)

  while queue:
    s = queue.pop(0) 
    print (s, end = " ") 

    for neighbour in graph[s]:
      if neighbour not in visited:
        visited.append(neighbour)
        queue.append(neighbour)


bfs(visited, graph, 'A')
print()